### ISSUE TITLE
Teddy bears should calm children

### ISSUE THREAD

**Issue by matshou** [cataclysmbnteam/Cataclysm-BN#1312]
# Is your feature request related to a problem? Please describe.

Teddy bear item currently has no use which makes it clutter.

# Describe the solution you'd like

Teddy bear should have the following effects on characters that are considered children:

- Help them fall asleep during night.
- Slightly increase general morale when present in inventory.
- Gives morale boost when interacted with (play with teddy bear).

# Describe alternatives you've considered

The teddy bear item can be removed from game.

# Additional context

Discussion on this topic started [here](https://github.com/cataclysmbnteam/Cataclysm-BN/issues/1309#issuecomment-1029694384).

**Comment by JonathanLochridge** [comment]
Currently I don't think having a child player character is supported? So for the inventory would that be a benefit that helps npcs only? I am not sure the status of how morale impacts npcs?
Or is having child characters meant to be an intended player option?

**Comment by Coolthulhu** [comment]
There are no human children in the game and they likely won't happen soon. Until then, teddy bears are fine to remove.

**Comment by Proxiehunter** [comment]
> Currently I don't think having a child player character is supported? So for the inventory would that be a benefit that helps npcs only? I am not sure the status of how morale impacts npcs? Or is having child characters meant to be an intended player option?

In addition I was under the impression they added to the comfort level while trying to get to sleep like having a proper bed and a pillow does, making it easier to fall asleep. Which is already a use.

**Comment by matshou** [comment]
> Currently I don't think having a child player character is supported? So for the inventory would that be a benefit that helps npcs only? I am not sure the status of how morale impacts npcs? Or is having child characters meant to be an intended player option?

My intent was only for player characters that are young enough to be considered children. Player characters under the age of 20 can be considered children in my opinion as many teenagers would revert to a childlike state when faced with a world shattering situation such as the Cataclym.

**Comment by matshou** [comment]
> > Currently I don't think having a child player character is supported? So for the inventory would that be a benefit that helps npcs only? I am not sure the status of how morale impacts npcs? Or is having child characters meant to be an intended player option?
> 
> In addition I was under the impression they added to the comfort level while trying to get to sleep like having a proper bed and a pillow does, making it easier to fall asleep. Which is already a use.

Ah you're right I haven't noticed the `SLEEP_AID` flag which makes them do exactly that:

```json
  {
    "type": "GENERIC",
    "id": "teddy",
    "symbol": "o",
    "color": "brown",
    "name": { "str": "teddy bear" },
    "description": "An old and half rotten teddy bear.  Looks like this one commemorates the grave of the child who once owned it.",
    "price": 200,
    "price_postapoc": 0,
    "material": [ "cotton" ],
    "weight": "514 g",
    "volume": "750 ml",
    "flags": [ "SLEEP_AID" ]
  }
  ```

**Comment by matshou** [comment]
> There are no human children in the game and they likely won't happen soon. Until then, teddy bears are fine to remove.

To be fair a human at the age of 16 (minimum in-game age) is pretty much a child in my opinion.

**Comment by Firestorm01X2** [comment]
Considering code SLEEP_AID already affecting characters. It is just needed to test if it affects player aswell.
https://github.com/cataclysmbnteam/Cataclysm-BN/blob/ae6e454d8264192529b50a1cb1a4c8b0c111ad2c/src/character.cpp#L5597

I guess it is already funtional

**Comment by chaosvolt** [comment]
The `SLEEP_AID` flag reasonably covers using a teddy bear for comfort, so what does that leave? The "morale source for children" listed in the OP wouldn't likely apply to most characters since we now have specific character ages, which have a minimum that's pretty solidly in the teenage range.

Back in the old days we had children player characters implemented as professions used for the school scenario, even including elementary students. This was purely flavor, having no bearing on character size, abilities, or how NPCs reacted to you in dialogue. Trying to bring child PCs back would basically require the ability to arbitrarily clamp character age and height to specific ranges, and allow that clamping to go below the normal minimum when required.

Technically, we COULD have a use for that ability even if we don't revive Bart Simpson's Mycus Adventure. There's still a high school student profession, and it's purely up to the player to not make them a senior citizen who's also a high school senior. Speaking of, there's also a senior citizen profession, which likewise would benefit from code support for preventing the player from making an anime vampire who's both 16 and somebody's grandpa.

That just leaves that last idea, which basically amounts to just adding a morale-boosting "play with item" action to teddy bears. This could in fact be applied to straw dolls and talking dolls too. Just like with teddy bears having `SLEEP_AID`, I don't think we really need to restrict that to currently-nonexistent child PCs, though we could opt to limit the usefulness of that option behind a trait, like Spiritual does for holy symbols.

For bonus points, if we dragged the various sportsball items out of obsoletion, we could add a craftable Wilson (actually being made from a volleyball sold separately, since that isn't yet one of said sportsball items), and if we got fairly complicated with our code for how doll/toy items generate morale, we could set it so that characters with the Psychosis trait get the most morale (and most interesting flavor messages) out of interacting with Wilsons. Related, playing with the "cursed" variant of the talking doll probably should give low or even negative morale unless you have the Psychopath trait.
